.github/workflows/semver-checks.yml

```diff
@@ -0,0 +1,39 @@
+on:   # yamllint disable-line rule:truthy
+  pull_request:
+
+name: Check semver breaks
+
+jobs:
+  API:
+    name: API - nightly toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # we need full history for cargo semver-checks
+      - name: "Install Rustup"
+        uses: dtolnay/rust-toolchain@nightly
```

Comment:
Oh, wait, this unpins the nightly. But let's merge this to unblock the CI and fix it afterwards.

Reply:
Something is odd here, we install the nightly toolchin but use the version in `nightly-version` in `contrib/check-semver.sh`. I'm surprised the script runs - unless we just happen to have the same nightly version as `dtonlnay`'s tool installs which I doubt?

Reply:
FTR still +1 from me to merge and fix later.

Reply:
Actually it's pinned in the script. This action is a bit misleading, we only need rustup on the machine.

Reply:
Yes, we can put anything in here. It's just to have rustup setup so that the next step does not fail.

Reply:
These steps

https://github.com/rust-bitcoin/rust-bitcoin/blob/f8c91b067b4851724fe68b3c7a41ec0181d04853/.github/workflows/semver-checks.yml#L17-L21

Reply:
Oh woops, missed that bit. Thanks